typescript/web/src/components/settings/workspace/profile.tsx

```diff
@@ -0,0 +1,198 @@
+import { useState, useEffect } from "react";
+import { ApolloError, gql, useMutation, useQuery } from "@apollo/client";
+import {
+  HStack,
+  Avatar,
+  useColorModeValue as mode,
+  Box,
+  SkeletonCircle,
+  Button,
+  Stack,
+  chakra,
+  StackDivider,
+  Input,
+  useToast,
+} from "@chakra-ui/react";
+import slugify from "slugify";
+import { RiGroupFill } from "react-icons/ri";
+import { useRouter } from "next/router";
+import {
+  Message,
+  searchWorkspacesQuery,
+} from "../../workspace-switcher/workspace-creation-modal";
+import { randomBackgroundGradient } from "../../../utils/random-background-gradient";
+import { FieldGroup, HeadingGroup, Card } from "..";
+
+const TeamIcon = chakra(RiGroupFill);
+
+const updateWorkspaceMutation = gql`
+  mutation updateWorkspace(
+    $workspaceSlug: String
+    $name: String
+    $image: String
+  ) {
+    updateWorkspace(
+      where: { slug: $workspaceSlug }
+      data: { name: $name, image: $image }
+    ) {
+      id
+      slug
+      name
+      image
+    }
+  }
+`;
+
+export const Profile = ({
+  workspace,
+  ...props
+}: {
+  workspace?: { id: string; name: string; image?: string | null; slug: string };
+}) => {
+  const router = useRouter();
+  const toast = useToast();
+  const [workspaceName, setWorkspaceName] = useState(workspace?.name ?? "");
+
+  const slug = slugify(workspaceName, { lower: true });
+
+  const [updateWorkspace, { error }] = useMutation<{
+    updateWorkspace: {
+      id: string;
+      name: string;
+      image?: string | null;
+      slug: string;
+    };
+  }>(updateWorkspaceMutation, {
+    variables: {
+      name: workspaceName !== workspace?.name ? workspaceName : null,
```

Comment:
Why do you do this check? It means that if at some point the save button is not disabled when `workspaceName === workspace?.name` then by clicking on save it will set the workspace name to `null`, not sure we want that

Reply:
True